src/block/carousel/index.php

```diff
@@ -19,3 +19,53 @@ function stackable_load_carousel_frontend_script() {
 	}
 	add_action( 'stackable/carousel/enqueue_scripts', 'stackable_load_carousel_frontend_script' );
 }
+
+// Lazy loaded images inside carousels make the carousel height buggy and show extra
+// spaces because the height isn't available in lazy loaded images. Prevent images from
+// hidden slides from lazy loading to prevent this.
+if ( ! function_exists( 'stackable_carousel_add_class_images' ) ) {
+	function stackable_carousel_add_class_images($block_content, $block) {
+		if ( ! class_exists( 'WP_HTML_Tag_Processor' ) ) {
+			return $block_content;
+		}
+
+		$index = 1;
+		$slides_to_show = 1;
+
+		if ( ! empty( $block ) && isset( $block['attrs'] ) && is_array( $block['attrs'] ) ) {
+			if ( isset( $block['attrs']['slidesToShow'] ) ) {
+				$slides_to_show = $block['attrs']['slidesToShow'];
+			}
+		}
+
+		$block_content = new WP_HTML_Tag_Processor( $block_content );
```

Comment:
I recommend using another variable name so that we do not accidentally mutate the original value, or use it mistakenly.

```suggestion
		$html_tag = new WP_HTML_Tag_Processor( $block_content );
```